internal/telemetry/collector.go

```diff
@@ -110,26 +97,47 @@ func (c *Collector) Collect(ctx context.Context) {
 	if err != nil {
 		glog.Errorf("Error exporting telemetry data: %v", err)
 	}
-	glog.V(3).Infof("Exported telemetry data: %x", traceData)
+	glog.V(3).Infof("Exported telemetry data: %+v", traceData)
 }
 
-// Start starts running NIC Telemetry Collector.
-func (c *Collector) Start(ctx context.Context) {
-	wait.JitterUntilWithContext(ctx, c.Collect, c.Period, 0.1, true)
+// BuildReport takes context and builds report from gathered telemetry data.
+func (c *Collector) BuildReport(ctx context.Context) (TraceData, error) {
+	dt := TraceData{}
+	var err error
+
+	for _, namespace := range c.Config.Namespaces {
+		dt.VirtualServers += c.GetVirtualServerCount(ctx, namespace)
+		dt.TransportServers += c.GetTransportServerCount(ctx, namespace)
+	}
+
+	return dt, err
 }
 
-// GetVSCount returns number of VirtualServers in watched namespaces.
-//
-// Note: this is a placeholder function.
-func (c *Collector) GetVSCount() int {
-	// Placeholder function
-	return 0
+// SetConfig is used to overwrite the existing config for the Collector.
+func (c *Collector) SetConfig(cfg CollectorConfig) {
```

Comment:
is this method used only to support a table tests, or we do have a case when we want to update the collector config at runtime?

Reply:
This was added to support the tests. It makes me think the tests could be structured better. Any thoughts on how to go about that?

Reply:
One test per one config sounds reasonable